lib/ansible/plugins/connection/ssh.py

```diff
@@ -692,7 +692,7 @@ def _build_command(self, binary, subsystem, *other_args):
             self._add_args(b_command, b_args, u"ansible.cfg set ssh_args")
 
         # Now we add various arguments that have their own specific settings defined in docs above.
-        if not self.get_option('host_key_checking'):
+        if self.get_option('host_key_checking') is False:
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, a test for this change would be good.

Reply:
this is not the correct change, add `default: True` to `host_key_checking` in `DOCUMENTATION` above

Reply:
Have added the change to the documentation and tested, is working as expected.  Revert line 696 or keep?  I think this is a better check as it's implicit for False rather than None/False, therefore edging on the view or checking keys unless implicitly turned off.  Happy to revert, just let me know :-)

Reply:
@Akasurde - changelog added as requested, thanks for the pointer